### PR TITLE
Injection Strategy docs

### LINKED ISSUE #1314: Documentation: Injection Strategy
With #1263 Constructor Injection has been added next to field injection. However, the current documentation lacks this feature.

@filiphr I can add the docs, where do you want me to put them (section)?

### PR DESCRIPTION
**Pull request by kevcodez**
Fixes #1314 

If you have additional information to put there or find a typo, let me know.

### CODE REVIEW

File: documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc [hunk 1]
```diff
@@ -690,6 +690,27 @@ private CarMapper mapper;
 
 A mapper which uses other mapper classes (see <<invoking-other-mappers>>) will obtain these mappers using the configured component model. So if `CarMapper` from the previous example was using another mapper, this other mapper would have to be an injectable CDI bean as well.
 
+[[injection-strategy]]
+=== Injection strategy
+
+When using <<using-dependency-injection,dependency injection>>, you can choose between field and constructor injection. This can be done by either providing the injection strategy via __Mapper__ or __MapperConfig__ annotation.
```
Comment (filiphr) [inline documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc]: I think using `@Mapper` and `@MapperConfig` with code styling would look better

File: documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc [hunk 2]
```diff
@@ -690,6 +690,27 @@ private CarMapper mapper;
...
+[subs="verbatim,attributes"]
+----
+@Mapper(componentModel = "cdi", uses = EngineMapper.class, injectionStrategy = InjectionStrategy.CONSTRUCTOR)
+public interface CarMapper {
+    CarDto carToCarDto(Car car);
+}
+----
+====
+
+The generated mapper will inject all classes defined in the **uses** attribute. When **InjectionStrategy.CONSTRUCTOR** is used, the constructor will have the appropriate annotation and the fields won't. When **InjectionStrategy.FIELD** is used, the annotation is on the field itself. For now, the default injection strategy is field injection. It is recommended to use constructor injection to simplify testing.
```
Comment (filiphr) [inline documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc]: Again I think that class should be in code style.

File: documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc [hunk 3]
```diff
@@ -690,6 +690,27 @@ private CarMapper mapper;
...
+
+If you inject additional fields by hand, we recommend using setter injection.
```
Comment (filiphr) [inline documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc]: I would make this a TIP. And I think that we should say that for abstract classes or decorators that setter injection should be used.

**Comment by filiphr** [comment]
Thanks a lot @kevcodez

**Comment by filiphr** [comment]
I also enhanced a bit in https://github.com/mapstruct/mapstruct/commit/3df399f69331e7ced44c89b5b40da6482df07d10. I hope you don't mind

**Comment by kevcodez** [comment]
@filiphr I don't mind at all, thanks for improving 👍

**Pull request merged**